ee/localserver/zta_auth_middleware.go

```diff
@@ -0,0 +1,285 @@
+package localserver
+
+import (
+	"crypto/ecdsa"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/lestrrat-go/jwx/jwk"
+)
+
+type ztaAuthMiddleware struct {
+	counterPartyKeys map[string]*ecdsa.PublicKey
+	slogger          *slog.Logger
+}
+
+type ztaResponse struct {
+	Data   []byte    `json:"data"`
+	PubKey *[32]byte `json:"pubKey"`
+}
+
+func (z *ztaAuthMiddleware) Wrap(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		boxParam := r.URL.Query().Get("payload")
+		if boxParam == "" {
+			z.slogger.Log(r.Context(), slog.LevelDebug,
+				"missing payload url param",
+			)
+
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		boxBytes, err := base64.URLEncoding.DecodeString(boxParam)
+		if err != nil {
+			z.slogger.Log(r.Context(), slog.LevelDebug,
+				"failed to decode payload",
+				"err", err,
+			)
+
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		var myChain chain
+		if err := json.Unmarshal(boxBytes, &myChain); err != nil {
+			z.slogger.Log(r.Context(), slog.LevelDebug,
+				"failed to unmarshal chain",
+				"err", err,
+			)
+
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		if err := myChain.validate(z.counterPartyKeys); err != nil {
+			z.slogger.Log(r.Context(), slog.LevelDebug,
+				"failed to validate chain",
+				"err", err,
+			)
+
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		newReq := &http.Request{
+			Method: http.MethodGet,
```

Comment:
Given this, I think we can maybe remove the `r.Method != http.MethodGet` check from `zta.go`? I guess fine to leave it, too